src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -315,6 +320,52 @@ class spell_mage_blazing_barrier : public AuraScript
     }
 };
 
+// 190356 - Blizzard
+// 4658 - AreaTrigger Create Properties
+struct areatrigger_mage_blizzard : AreaTriggerAI
+{
+    areatrigger_mage_blizzard(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger), _refreshTimer(TICK_PERIOD) { }
+
+    static constexpr Milliseconds TICK_PERIOD = Milliseconds(1000);
+
+    void OnUpdate(uint32 diff) override
+    {
+        _refreshTimer -= Milliseconds(diff);
+
+        while (_refreshTimer <= 0s)
+        {
+            if (Unit* caster = at->GetCaster())
+                caster->CastSpell(at->GetPosition(), SPELL_MAGE_BLIZZARD_DAMAGE);
+
+            _refreshTimer += TICK_PERIOD;
+        }
+    }
+
+private:
+    Milliseconds _refreshTimer;
```

Comment:
we arent refreshing anything here, suggesting `_tickTimer`

Reply:
Accepted!